### PR TITLE
Support tso service

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### CODE REVIEW

**Comment by rleungx** [comment]
@binshi-bing Would you like to take a look?

File: proto/tsopb.proto [hunk 1]
```diff
@@ -0,0 +1,42 @@
+syntax = "proto3";
+package tsopb;
```
Comment (binshi-bing) [inline proto/tsopb.proto:2]: According to our last sync, my understanding is that we'll provide these APIs as storage interface used by tso service internal.

File: proto/tsopb.proto [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import "gogoproto/gogo.proto";
+import "rustproto.proto";
+import "pdpb.proto";
+
+option (gogoproto.sizer_all) = true;
+option (gogoproto.marshaler_all) = true;
+option (gogoproto.unmarshaler_all) = true;
+option (rustproto.lite_runtime_all) = true;
+
+service Tso {
```
Comment (binshi-bing) [inline proto/tsopb.proto:13]: This service is to provide TSO storage. Shall we call it TsoStorage to differentiate it from the TSO service which provide GetTSs?

**Review by binshi-bing — COMMENTED** [review]
Though we don't need this change for now, this is my first review for PD.

**Pull request closed without merging**